pulp_smash/tests/rpm/api_v2/test_noop.py

```diff
@@ -0,0 +1,264 @@
+3. Remove unit X from repository foo and re-publish.
+4. Assert unit X is not accessible.
+"""
+from __future__ import unicode_literals
+
+import unittest2
+
+from pulp_smash import api, utils, selectors
+from pulp_smash.compat import urljoin
+from pulp_smash.constants import REPOSITORY_PATH, RPM_FEED_URL
+from pulp_smash.tests.rpm.api_v2.utils import (
+    gen_distributor,
+    gen_repo,
+    xml_handler,
+    NAMESPACE
+)
+from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+_PUBLISH_DIR = 'pulp/repos/'
+
+
+def get_repo_md(client, distributor, ):
+    """Return parsed repomd.xml file."""
+    client.response_handler = xml_handler
+    path = urljoin('/pulp/repos/', distributor['config']['relative_url'])
+    path = urljoin(path, 'repodata/repomd.xml')
+    root_elem = client.get(path)
+    client.response_handler = api.json_handler
+    return root_elem
+
+
+def get_timestamps(root_elem):
+    """Return dictionary of timestamps for each data entry in repomd."""
+    data_elems = root_elem.findall('{%s}data' % NAMESPACE)
+    timestamps = {}
+    for data in data_elems:
+        stamp = data.find('{%s}timestamp' % NAMESPACE).text
+        timestamps[data.attrib.get('type')] = stamp
+    return timestamps
+
+
+class NOOPTestCase1(utils.BaseAPITestCase):
+    """Test NOOP publish on rpm repo with overrides_config."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create one repository with feed, unassociate unit and re-publish.
+
+        Following steps are executed:
+
+        1. Create repository foo with feed, sync and publish it.
+        2. assert publish is operational
+        3. publish repository again
+        4. assert publish is noop
+        5. publish with overrides_config
+        6. assert publish is operational
+        7. publish with overrides_config (identical to step 5)
+        8. assert publish is noop
+
+        """
+        super(NOOPTestCase1, cls).setUpClass()
+        cls.responses = {}
+        cls.tasks = {}
+        cls.xmldata = {}
+        if selectors.bug_is_untestable(1724, cls.cfg.version):
+            raise unittest2.SkipTest('https://pulp.plan.io/issues/1724')
+
+        # Create and sync a repository.
+        client = api.Client(cls.cfg)
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_FEED_URL
+        repo_href = client.post(REPOSITORY_PATH, body).json()['_href']
+        cls.resources.add(repo_href)  # mark for deletion
+        cls.responses['sync'] = utils.sync_repo(cls.cfg, repo_href)
+
+        # Add a distributor and publish it.
+        distributor = client.post(
+            urljoin(repo_href, 'distributors/'),
+            gen_distributor(),
+        ).json()
+        cls.responses['first publish'] = client.post(
+            urljoin(repo_href, 'actions/publish/'),
+            {'id': distributor['id']},
+        ).json()
+        next(api.poll_spawned_tasks(cls.cfg, cls.responses['first publish']))
```

Comment:
By default, the `Client` class polls all spawned tasks and waits for them to complete. These `next` statements read all of the tasks spawned by `cls.responses['first publish']`. Given that all of the tasks are already complete, this is redundant.

You don't need to fix this if you don't want. But this is the sort of thing I'll be changing in a follow-up commit.